deploy/deploy.ts

```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
 
   dim(`deployer: ${admin}`);
 
-  cyan(`\nDeploying ATokenYieldSourceProxyFactory...`);
-  const aTokenYieldSourceProxyFactoryResult = await deploy('ATokenYieldSourceProxyFactory', {
+  cyan(`\nDeploying ATokenYieldSource...`);
+  const aTokenYieldSourceResult = await deploy('ATokenYieldSource', {
     from: deployer,
+    skipIfAlreadyDeployed: true
   });
-  displayResult('ATokenYieldSourceProxyFactory', aTokenYieldSourceProxyFactoryResult);
+  
+  displayResult('ATokenYieldSource', aTokenYieldSourceResult);
+
+  let aTokenYieldSourceContract
+  if(aTokenYieldSourceResult.address){
```

Comment:
I don't think you need the conditional.  The aTokenYieldSourceResult should always be defined

Reply:
If i define types on everything then yes!